src/cascadia/TerminalApp/Jumplist.cpp

```diff
@@ -149,19 +149,39 @@ winrt::com_ptr<IShellLinkW> Jumplist::_createShellLink(const std::wstring_view n
     const auto module{ GetWtExePath() };
     THROW_IF_FAILED(sh->SetPath(module.data()));
     THROW_IF_FAILED(sh->SetArguments(args.data()));
+    auto propStore{ sh.as<IPropertyStore>() };
 
     PROPVARIANT titleProp;
     titleProp.vt = VT_LPWSTR;
     titleProp.pwszVal = const_cast<wchar_t*>(name.data());
 
-    PROPVARIANT iconProp;
-    iconProp.vt = VT_LPWSTR;
-    iconProp.pwszVal = const_cast<wchar_t*>(path.data());
+    // Check for a comma in the path. If we find one we have an indirect icon. Parse the path into a file path and index/id.
+    auto commaPosition = path.find(L",");
+    if (commaPosition != std::wstring_view::npos)
+    {
+        const std::wstring iconPath{ path.substr(0, commaPosition) };
+
+        // We dont want the comma included so add 1 to its position
+        const std::wstring iconIndexString{ path.substr(commaPosition + 1) };
+        std::wstringstream stringStream;
+        stringStream << iconIndexString;
+        int iconIndex = 0;
+        stringStream >> iconIndex;

```

Comment:
I'd prefer we avoid stringstreams here - it's not a hot path, but it does establish a precedent. Is `til::to_ulong` insufficient? Should we fix it?

Reply:
If you want to support icon indexes ie -4 it needs to support signed numbers. I didn't see a to_* that supported signed numbers. I had a feeling this but would be a bit controversial 🙃

Reply:
https://devblogs.microsoft.com/oldnewthing/20100505-00/?p=14153

this post was where I gathered that intel. If you just want to support positive numbers I can revert that stuff. Either way is good with me ofc.

Reply:
OH DUH SORRY. I didn't read, clearly.

I'd rather we introduce something like `til::to_long` nearby `til::to_ulong` to keep the dependency surface small. Heck, you could even implement it as "if there's a `-` at the beginning, strip it off and call `til::to_ulong`"

(though perhaps at that point it would be `til::to_int` (because we would lose some bits by casting down from ulong to long, and we wouldn't want to lie))

Reply:
Ok. I will work on that ☺️

Reply:
> Okay, now, silly question: does it work for `"icon": "%SystemRoot%\\System32\\liveCaptions.exe"`?

We could do something similar to the winui side and default to ,0 if no index is given. That would fix this weird one.

Reply:
nvm cant do that  cause it breaks all of the other profiles in the jump list

<img width="291" alt="image" src="https://user-images.githubusercontent.com/2086722/234429623-334d8955-261b-41ad-a5f5-30c9f910ca5f.png">

Reply:
Could special case that one.

Reply:
Oh.. I lied. The others are not exe resources.. hmm

Reply:
Heyyyyy.. On the jumplist side we can check if the path ends with exe or dll (or any other file ext that you want really that could contain an embedded resource) and if so default to 0 for icon index otherwise do what we've always done. 

<img width="978" alt="image" src="https://user-images.githubusercontent.com/2086722/234430546-7d683b31-66ad-4020-8240-656289acafa2.png">


Sorry for the spam. :|